Simplified/Simplified.js

```diff
@@ -188,6 +188,31 @@ function Simplified() {
       }";
     innerDocument.head.appendChild(styleElement);
   }
+  
+  /**
+   * setCfiBeforeChange
+   * Before there's a font size or font family change in the UI, keep track of the CFI of
+   * the first visible element in the viewport.
+   */
+  this.setCfiBeforeChange = function() {
+    var firstVisibleCfi = ReadiumSDK.reader.getFirstVisibleCfi();
+    this.cfiBeforeChange = firstVisibleCfi;
+  }
+
+  /**
+   * updateCFI
+   * If there's an existing CFI that we are tracking, open the reader to that CFI.
+   * TODO: the CFI works well for the previous font size or font family. When switching to
```

Comment:
Yes, I have long thought that this was the endgame. I would also like to include a CFI that is "near" (but unambiguously before) the text, as a fallback and as a way to speed up the search.

That would be analogous to opening a book to the page marked by a bookmark, then scanning the page to find where you left off reading. I think that's the most reliable way of doing this.

Can we do this in this branch? I didn't mention it earlier because I assumed it was too complicated for Edwin's first job. But 100% we should move in this direction.

Reply:
I think a CFI can include nearby text. Certainly people have talked about that; I don't know if it made it into the spec (or if there even is a spec).

If you think of the text as _part_ of the CFI then you're still generating a CFI and then navigating to it. The generation and navigation processes are a little more complex, but it's not a totally new action.

Reply:
Right, the CFI points to a specific text node, but I wasn't sure if that's what Greg meant about storing the text snippet. I'll look up if CFI can include nearby text in a sibling node or something.

Reply:
Disregard my question for now and we can come back to it.